File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        HashSet<Integer> hmap = new HashSet<>();
        for(int num:nums){
            hmap.add(num);
        }
        if(hmap.size()>nums.length/2){
            return nums.length/2;
        }else{
            return hmap.size();
        }
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <map>
using namespace std;

int solution(vector<int> nums)
{
    int answer = 0;
    int count=nums.size()/2;
    map<int,int> map1;
    for(int i=0;i<nums.size();i++){
        if(map1.find(nums[i])!=map1.end()){     // map에 존재하면..(중복건이면)
        }else{
            map1.insert({nums[i],1});
            answer++;
        }
    }
 
    if(answer>=count){
        return count;
    }
    return map1.size();
    
    
    return answer;
}